invenio_rdm_records/services/components.py

```diff
@@ -144,22 +146,73 @@ def new_version(self, identity, draft=None, record=None, **kwargs):
 class ExternalPIDsComponent(ServiceComponent):
     """Service component for pids."""
 
+    def get_provider(self, scheme):
+        """Process a given PID with its provider."""
+        provider = self.service.config.pids_providers.get(scheme)
+
+        if not provider:
+            raise ValidationError(
```

Comment:
Here I would raise more a 500, because it is a misconfiguration of the system and not a 400, a wrong user input.
So raise maybe a `KeyError`? Generic `RDM Expection` if any?

Reply:
there is no generic that I know of, so I'm changing the accessor to [] so it KeyErrors if so.